terraform/stacks/tooling/cross_account_iam.tf

```diff
@@ -0,0 +1,54 @@
+resource "aws_iam_role" "payer_account_access" {
+    name = "payer-account-access"
+
+    assume_role_policy = <<EOF
+    {
+      "Version": "2012-10-17",
+      "Statement": [
+        {
+          "Effect": "Allow",
+          "Principal": {
+            "AWS": "arn:aws:iam::${var.payer_account_id}:root"
+          },
+          "Action": "sts:AssumeRole",
+          "Condition": {
+            "Bool": {
+              "aws:MultiFactorAuthPresent": "true"
+            }
+          }
+        }
+      ]
+    }
+    EOF
+}
+
+resource "aws_iam_policy" "read_s3_tags" {
+    name = "read-s3-bucket-tags"
+    description = "Allow reading bucket tags in s3"
+
+    policy = <<EOF
+    {
+      "Version": "2012-10-17",
+      "Statement": [
+        {
+          "Effect": "Allow",
+          "Action": "s3:ListAllMyBuckets",
+          "Resource": "*"
+        },
+        {
+          "Effect": "Allow",
+          "Action": [
+            "s3:GetBucketLocation",
+            "s3:GetBucketTagging"
+           ],
+          "Resource": "arn:aws:s3:::*"
+        }
+      ]
+    }
+    EOF
+}
```

Comment:
Just confirming:  the payer account only needs to read S3 bucket tags and that's it?  No need for writing tags or anything else?

Reply:
yeah - the payer account needs to enable cost allocation tracking on some tags we're applying. The trick is the cost allocation tracking is a feature of their account, but relies on seeing the tags in our account.